libraries/core_libs/consensus/include/pbft/pbft_manager.hpp

```diff
@@ -601,11 +624,13 @@ class PbftManager : public std::enable_shared_from_this<PbftManager> {
   size_t step_ = 1;
   size_t startingStepInRound_ = 1;
 
-  std::optional<std::pair<blk_hash_t, uint64_t /* period */>> soft_voted_block_for_round_{};
+  // Block that node saw 2t+1 soft votes for
+  std::optional<std::pair<blk_hash_t, std::shared_ptr<PbftBlock>>> soft_voted_block_for_round_{};
 
-  std::optional<std::pair<blk_hash_t, uint64_t /* period */>> cert_voted_block_for_round_{};
+  // Block that node cert voted
+  std::optional<std::shared_ptr<PbftBlock>> cert_voted_block_for_round_{};
```

Comment:
Isn't this optional excess here?

Reply:
you are probably right. I will create another PR https://github.com/Taraxa-project/taraxa-node/issues/2034